pkg/protostore/protostore.go

```diff
@@ -413,12 +440,69 @@ func (p ProtobufDataStore) FindAll(ctx context.Context, msgs interface{}, pagina
 	return metadataMap, nil
 }
 
-func (p ProtobufDataStore) SoftDeleteById(ctx context.Context, id string, msg proto.Message) (int64, error) {
+func (p ProtobufDataStore) FindAllIncludingSoftDeleted(ctx context.Context, msgs interface{}, pagination *datastore.Pagination) (metadataMap map[string]Metadata, err error) {
```

Comment:
Do you need replicate the whole method? or can create helper method?

Reply:
Replication for backward compatibility. Or add a "softDelete" flag directly inside FindAllIncludingSoftDeleted() but that modifies function signatures for all usages

Reply:
change pushed. Please check latest